pkg/cmd/repo/edit/edit.go

```diff
@@ -0,0 +1,120 @@
+package edit
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/spf13/cobra"
+)
+
+type EditOptions struct {
+	HTTPClient *http.Client
+	Repository ghrepo.Interface
+	Edits      EditRepositoryInput
+}
+
+type EditRepositoryInput struct {
+	Name                *string `json:"name,omitempty"`
+	Description         *string `json:"description,omitempty"`
+	Homepage            *string `json:"homepage,omitempty"`
+	EnableIssues        *bool   `json:"has_issues,omitempty"`
+	EnableProjects      *bool   `json:"has_projects,omitempty"`
+	EnableWiki          *bool   `json:"has_wiki,omitempty"`
+	IsTemplate          *bool   `json:"is_template,omitempty"`
+	DefaultBranch       *string `json:"default_branch,omitempty"`
+	EnableSquashMerge   *bool   `json:"allow_squash_merge,omitempty"`
+	EnableMergeCommit   *bool   `json:"allow_merge_commit,omitempty"`
+	EnableRebaseMerge   *bool   `json:"allow_rebase_merge,omitempty"`
+	EnableAutoMerge     *bool   `json:"allow_auto_merge,omitempty"`
+	DeleteBranchOnMerge *bool   `json:"delete_branch_on_merge,omitempty"`
+}
+
+func NewCmdEdit(f *cmdutil.Factory, runF func(options *EditOptions) error) *cobra.Command {
+	opts := &EditOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "edit [<repository>]",
+		Short: "Edit repository settings",
+		Annotations: map[string]string{
+			"help:arguments": heredoc.Doc(`
+				A repository can be supplied as an argument in any of the following formats:
+				- "OWNER/REPO"
+				- by URL, e.g. "https://github.com/OWNER/REPO"
+			`),
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.Flags().NFlag() == 0 {
+				return cmdutil.FlagErrorf("at least one flag is required")
+			}
+
+			if len(args) > 0 {
+				var err error
+				opts.Repository, err = ghrepo.FromFullName(args[0])
+				if err != nil {
+					return err
+				}
+			} else {
+				var err error
+				opts.Repository, err = f.BaseRepo()
+				if err != nil {
+					return err
+				}
+			}
+
+			if httpClient, err := f.HttpClient(); err == nil {
+				opts.HTTPClient = httpClient
+			} else {
+				return err
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return editRun(opts)
+		},
+	}
+
+	cmdutil.NilStringFlag(cmd, &opts.Edits.Name, "name", "", "Change the name of the repository")
```

Comment:
One nit is that we can probably change the description to be "Name of the repository" as it is already in the context of the `repo edit` command.